core/account-id/src/lib.rs

```diff
@@ -64,49 +281,34 @@ impl AccountId {
         }
     }
 
-    /// Creates an AccountId without any validation
-    ///
-    /// Useful in cases where pre-validation causes unforseen issues.
+    /// Creates an `AccountId` without any validation checks.
     ///
-    /// Note: this is restrictively for internal use only, and, being behind a feature flag,
+    /// Please note that this is restrictively for internal use only. Plus, being behind a feature flag,
     /// this could be removed later in the future.
     ///
-    /// # Safety
+    /// ## Safety
+    ///
+    /// Since this skips validation and constructs an `AccountId` regardless,
+    /// the caller bears the responsibility of ensuring that the Account ID is valid.
+    /// You can use the [`AccountId::validate`] function sometime after its creation but before it's use.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// use near_account_id::AccountId;
     ///
-    /// You must ensure to manually call the [`AccountId::validate`] function on the
-    /// AccountId sometime after its creation but before it's use.
+    /// let alice = AccountId::new_unvalidated("alice.near".to_string());
+    /// assert!(AccountId::validate(alice.as_ref()).is_ok());
+    ///
+    /// let ƒelicia = AccountId::new_unvalidated("ƒelicia.near".to_string());
+    /// assert!(AccountId::validate(ƒelicia.as_ref()).is_err());
+    /// ```
     #[cfg(feature = "internal_unstable")]
     #[deprecated(since = "#4440", note = "AccountId construction without validation is illegal")]
     pub fn new_unvalidated(account_id: String) -> Self {
         Self(account_id.into())
     }
 
-    pub fn is_top_level_account_id(&self) -> bool {
-        self.len() >= MIN_ACCOUNT_ID_LEN
-            && self.len() <= MAX_ACCOUNT_ID_LEN
-            && self.as_ref() != "system"
-            && !self.as_ref().contains('.')
-    }
-
-    /// Returns true if the signer_id can create a direct sub-account with the given account Id.
-    pub fn is_sub_account_of(&self, parent_account_id: &AccountId) -> bool {
-        if parent_account_id.len() >= self.len() {
-            return false;
-        }
-        // Will not panic, since valid account id is utf-8 only and the length is checked above.
-        // e.g. when `near` creates `aa.near`, it splits into `aa.` and `near`
-        let (prefix, suffix) = self.0.split_at(self.len() - parent_account_id.len());
-
-        prefix.find('.') == Some(prefix.len() - 1) && suffix == parent_account_id.as_ref()
-    }
-
-    /// Returns true if the account ID length is 64 characters and it's a hex representation.
-    pub fn is_implicit(account_id: &str) -> bool {
-        account_id.len() == 64
-            && account_id.as_bytes().iter().all(|b| matches!(b, b'a'..=b'f' | b'0'..=b'9'))
-    }
-
-    /// Returns true if the account ID is the system account.
     pub fn is_system(&self) -> bool {
         self.as_ref() == "system"
     }
```

Comment:
Bellow it seems better to delete `test_account` function. I don't think that `test` has any special meaning for the protocol, so there shouldn't be a random fn to create that specific naem.

Reply:
Bellow, why it is 

```
impl<T: ?Sized> AsRef<T> for AccountId
where
    Box<str>: AsRef<T>,
```

and not just 

```
impl AsRef<str> for AccountId
```

?

Reply:
Bellow, in the Display impl, it's bettre to use `fmt::Display::fmt` rather than `write!` to get support for padding for free: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=464ccb8bc9a815d9c0193e742af51908

Reply:
Bellow, we have `impl From<AccountId> for Box<str>` but not the opposite `TryFrom<Box<str>>`.

Reply:
In addition to `AsRef`, I'd probably add

```rust
impl AccountId {
  pub fn as_str(&self) -> &str
}
```